python/lsst/atmospec/nightAnalysis.py

```diff
@@ -78,56 +73,61 @@ def getLineValue(table, lineName, columnName, nanForMissingValues=True):
     return table[columnName][rowNum]
 
 
-class NightStellarSpectra():
+class NightStellarSpectra:
     """Class for holding the results for a night's observations of a given star
     """
 
-    def __init__(self, rerunDir, dayObs, targetName, *, butler=None, ignoreSeqNums=[]):
+    def __init__(self, rerunDir, dayObs, targetName, instrument, *, butler=None, ignoreSeqNums=[],
+                 collections=None):
         self.rerunDir = rerunDir
-        self.dayObs = dayObs
+        self.dayObs = int(dayObs)
         self.targetName = targetName
+        if instrument is None:
+            raise ValueError("An instrument name must be provided.")
+        self.instrument = instrument
+
         if butler:
-            self.butler = butler
+            self.butler = dafButler.Butler(butler=butler, instrument=instrument, collections=collections)
         else:
-            self.butler = dafPersist.Butler(rerunDir)
+            self.butler = dafButler.Butler(rerunDir, instrument=instrument, collections=collections)
```

Comment:
I think we can hard-code `instrument` as `'LATISS'` everywhere here - this entire package is explicitly AuxTel-only. That will also mean that we can simply remove `rerunDir`, which is now not a `rerunDir` at all, but in fact the butler root, now that this is Gen3.

Reply:
As discussed on slack, I will retain an optional instrument parameter for the constructor but have it default to LATISS. That way at least some enterprising soul has the option of experimenting with some other spectrograph. The code no longer requires certain headers to exist.